slsim/LsstSciencePipeline/util_lsst.py

```diff
@@ -210,3 +214,373 @@ def opsim_variable_lens_injection(
+                    symmetric=symmetric,
+                )
+            )
+            magnitude_image_4.append(mag_gaussian_realiz_4)
+            mag_error_4_minus.append(mag_err_4_minus)
+            mag_error_4_plus.append(mag_err_4_plus)
+        if lens_class.image_number[0] == 2:
+            magnitude_image_3.append(-1)
+            mag_error_3_minus.append(-1)
+            mag_error_3_plus.append(-1)
+            magnitude_image_4.append(-1)
+            mag_error_4_minus.append(-1)
+            mag_error_4_plus.append(-1)
+    # Create a astropy table of transient data.
+    lens_id_col = Column(name="lens_id", data=np.array(lens_id).squeeze())
+    magnitude_col_image_1 = Column(
+        name="mag_image_1", data=np.array(magnitude_image_1).squeeze()
+    )
+    magnitude_col_image_2 = Column(
+        name="mag_image_2", data=np.array(magnitude_image_2).squeeze()
+    )
+    magnitude_col_image_3 = Column(
+        name="mag_image_3", data=np.array(magnitude_image_3).squeeze()
+    )
+    magnitude_col_image_4 = Column(
+        name="mag_image_4", data=np.array(magnitude_image_4).squeeze()
+    )
+    total_mag_error_col_1_low = Column(
+        name="mag_error_image_1_low", data=np.array(mag_error_1_minus).squeeze()
+    )
+    total_mag_error_col_1_high = Column(
+        name="mag_error_image_1_high", data=np.array(mag_error_1_plus).squeeze()
+    )
+    total_mag_error_col_2_low = Column(
+        name="mag_error_image_2_low", data=np.array(mag_error_2_minus).squeeze()
+    )
+    total_mag_error_col_2_high = Column(
+        name="mag_error_image_2_high", data=np.array(mag_error_2_plus).squeeze()
+    )
+    total_mag_error_col_3_low = Column(
+        name="mag_error_image_3_low", data=np.array(mag_error_3_minus).squeeze()
+    )
+    total_mag_error_col_3_high = Column(
+        name="mag_error_image_3_high", data=np.array(mag_error_3_plus).squeeze()
+    )
+    total_mag_error_col_4_low = Column(
+        name="mag_error_image_4_low", data=np.array(mag_error_4_minus).squeeze()
+    )
+    total_mag_error_col_4_high = Column(
+        name="mag_error_image_4_high", data=np.array(mag_error_4_plus).squeeze()
+    )
+    image_list_col = Column(name="lens_image", data=image_list)
+    # add these new quantities as new columns in the expo_data table.
+    copied_exposure_data.add_columns(
+        [
+            lens_id_col,
+            magnitude_col_image_1,
+            magnitude_col_image_2,
+            magnitude_col_image_3,
+            magnitude_col_image_4,
+            total_mag_error_col_1_low,
+            total_mag_error_col_1_high,
+            total_mag_error_col_2_low,
+            total_mag_error_col_2_high,
+            total_mag_error_col_3_low,
+            total_mag_error_col_3_high,
+            total_mag_error_col_4_low,
+            total_mag_error_col_4_high,
+            image_list_col,
+        ]
+    )
+    return copied_exposure_data
+
+
+def extract_lightcurves_in_different_bands(data, images=True):
```

Comment:
is this definition (and the other one above) specific to LSST? Perhaps you can move this definition to another file